compute/client_library/snippets/disks/attach_disk.py

```diff
@@ -0,0 +1,120 @@
+    If the operation ends with an error, an exception will be raised.
+    If there were any warnings during the execution of the operation
+    they will be printed to sys.stderr.
+
+    Args:
+        operation: a long-running operation you want to wait on.
+        verbose_name: (optional) a more verbose name of the operation,
+            used only during error and warning reporting.
+        timeout: how long (in seconds) to wait for operation to finish.
+            If None, wait indefinitely.
+
+    Returns:
+        Whatever the operation.result() returns.
+
+    Raises:
+        This method will raise the exception received from `operation.exception()`
+        or RuntimeError if there is no exception set, but there is an `error_code`
+        set for the `operation`.
+
+        In case of an operation taking longer than `timeout` seconds to complete,
+        a `concurrent.futures.TimeoutError` will be raised.
+    """
+    result = operation.result(timeout=timeout)
+
+    if operation.error_code:
+        print(
+            f"Error during {verbose_name}: [Code: {operation.error_code}]: {operation.error_message}",
+            file=sys.stderr,
+            flush=True,
+        )
+        print(f"Operation ID: {operation.name}", file=sys.stderr, flush=True)
+        raise operation.exception() or RuntimeError(operation.error_message)
+
+    if operation.warnings:
+        print(f"Warnings during {verbose_name}:\n", file=sys.stderr, flush=True)
+        for warning in operation.warnings:
+            print(f" - {warning.code}: {warning.message}", file=sys.stderr, flush=True)
+
+    return result
+
+
+def attach_disk(
+    project_id: str,
+    zone: str,
+    instance_name: str,
+    disk_link: str,
+    mode: str = "READ_ONLY",
+) -> NoReturn:
+    """
+    Attaches a non-boot persistent disk to a specified compute instance. The disk might be zonal or regional.
+
+    You need following permissions to execute this action:
+    https://cloud.google.com/compute/docs/disks/regional-persistent-disk#expandable-1
+
+    Args:
+        project_id: project ID or project number of the Cloud project you want to use.
+        zone:name of the zone in which the instance you want to use resides.
+        instance_name: name of the compute instance you want to attach a disk to.
+        disk_link: full or partial URL to a persistent disk that you want to attach. This can be either
+            regional or zonal disk.
+            Expected formats:
+                * https://www.googleapis.com/compute/v1/projects/[project]/zones/[zone]/disks/[disk_name]
+                * /projects/[project]/zones/[zone]/disks/[disk_name]
+                * /projects/[project]/regions/[region]/disks/[disk_name]
+        mode: Specifies in what mode the disk will be attached to the instance. Available options are `READ_ONLY`
+            and `READ_WRITE`. Disk in `READ_ONLY` mode can be attached to multiple instances at once.
```

Comment:
Could you specify why we default to `READ_ONLY` if that's needed? If there's no reason, we should omit having a default value.